framework/src/media/audio/audio_manager.c

```diff
@@ -952,18 +1035,52 @@ int start_audio_stream_out(void *data, unsigned int frames)
 		medvdbg("Resume the output audio card!!\n");
 	}
 
-	if (card->resample.necessary) {
-		resampled_frames = resample_stream_out(card, data, frames);
+	if ((device_channel_num = pcm_get_channels(card->pcm)) == 0) {
+		meddbg("Fail to get channel number\n");
+		goto error_with_lock;
+	}
+	rechanneling_ratio = (float)device_channel_num / (float)card->resample.user_channel;
+
+	if (card->resample.ratio != 1) {
+		if (rechanneling_ratio > 1) {	// Mono -> Stereo
+			char *src = (char *)data;
+			char *dst = (char *)card->resample.rechanneling_buffer;
+			medvdbg("Start addr = %x\n", card->resample.rechanneling_buffer);
+			for (int i = 0; i < frames; i++) {
+				for (int j = 0; j < 2; j++) {
+					dst[i * 2 * 2 + j] = src[i * 2 + j];
+					dst[i * 2 * 2 + 2 + j] = src[i * 2 + j];
+				}
+			}
+			medvdbg("   Rechanneling, Buffer Size = %d, channel ratio: %f\n", card->resample.rechanneling_buffer_size, rechanneling_ratio);
```

Comment:
you can remove the extra space start of debug logs

Reply:
Modified.